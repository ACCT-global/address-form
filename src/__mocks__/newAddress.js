export default {
  addressId: '1',
  addressType: 'residential',
  city: null,
  complement: null,
  country: null,
  geoCoordinates: null,
  neighborhood: null,
  number: null,
  postalCode: null,
  receiverName: null,
  reference: null,
  state: null,
  street: null,
}
