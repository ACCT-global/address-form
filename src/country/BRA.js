import { POSTAL_CODE } from '../constants'

export default {
  country: 'BRA',
  abbr: 'BR',
  postalCodeFrom: POSTAL_CODE,
  postalCodeProtectedFields: ['state', 'city'],
  fields: [
    {
      name: 'postalCode',
      fixedLabel: 'CEP',
      required: true,
      mask: '99999-999',
      regex: '^([\\d]{5})\\-?([\\d]{3})$',
      postalCodeAPI: true,
      forgottenURL:
        'http://www.buscacep.correios.com.br/servicos/dnec/index.do',
      size: 'small',
    },
    {
      name: 'street',
      label: 'street',
      required: true,
      size: 'xlarge',
    },
    {
      name: 'number',
      label: 'number',
      required: true,
      size: 'mini',
    },
    {
      name: 'complement',
      label: 'complement',
      size: 'large',
    },
    {
      name: 'neighborhood',
      label: 'neighborhood',
      required: true,
      size: 'large',
    },
    {
      name: 'city',
      label: 'city',
      required: true,
      size: 'large',
    },
    {
      name: 'state',
      label: 'state',
      required: true,
      size: 'medium',
      optionsCaption: 'UF',
      optionsPairs: [
        { value: 'AC', label: 'Acre' },
        { value: 'AL', label: 'Alagoas' },
        { value: 'AP', label: 'Amapá' },
        { value: 'AM', label: 'Amazonas' },
        { value: 'BA', label: 'Bahia' },
        { value: 'CE', label: 'Ceará' },
        { value: 'DF', label: 'Distrito Federal' },
        { value: 'ES', label: 'Espírito Santo' },
        { value: 'GO', label: 'Goiás' },
        { value: 'MA', label: 'Maranhão' },
        { value: 'MT', label: 'Mato Grosso' },
        { value: 'MS', label: 'Mato Grosso do Sul' },
        { value: 'MG', label: 'Minas Gerais' },
        { value: 'PA', label: 'Pará' },
        { value: 'PB', label: 'Paraíba' },
        { value: 'PR', label: 'Paraná' },
        { value: 'PE', label: 'Pernambuco' },
        { value: 'PI', label: 'Piauí' },
        { value: 'RJ', label: 'Rio de Janeiro' },
        { value: 'RN', label: 'Rio Grande do Norte' },
        { value: 'RS', label: 'Rio Grande do Sul' },
        { value: 'RO', label: 'Rondônia' },
        { value: 'RR', label: 'Roraima' },
        { value: 'SC', label: 'Santa Catarina' },
        { value: 'SP', label: 'São Paulo' },
        { value: 'SE', label: 'Sergipe' },
        { value: 'TO', label: 'Tocantins' },
      ],
    },
  ],
  geolocation: {
    postalCode: {
      valueIn: 'long_name',
      types: ['postal_code'],
      required: false,
    },
    number: {
      valueIn: 'long_name',
      types: ['street_number'],
      required: false,
    },
    street: {
      valueIn: 'long_name',
      types: ['route'],
      required: false,
    },
    neighborhood: {
      valueIn: 'long_name',
      types: [
        'neighborhood',
        'sublocality_level_1',
        'sublocality_level_2',
        'sublocality_level_3',
        'sublocality_level_4',
        'sublocality_level_5',
      ],
      required: false,
    },
    state: {
      valueIn: 'short_name',
      types: ['administrative_area_level_1'],
      required: false,
    },
    city: {
      valueIn: 'long_name',
      types: ['administrative_area_level_2', 'locality'],
      required: false,
    },
  },
}
