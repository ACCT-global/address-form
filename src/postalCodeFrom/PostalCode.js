import React, { Component } from 'react'
import PropTypes from 'prop-types'
import AddressShapeWithValidation
  from '../propTypes/AddressShapeWithValidation'
import { getField } from '../selectors/fields'
import InputLabel from '../addressInputs/InputLabel'
import InputText from '../addressInputs/InputText'
import PostalCodeLoader from './PostalCodeLoader'

class PostalCode extends Component {
  handleChange = value => {
    const { address, onChangeAddress } = this.props

    onChangeAddress({
      postalCode: {
        ...address.postalCode,
        autoCompleted: undefined,
        value,
      },
    })
  };

  handleBlur = () => {
    const { address, onChangeAddress } = this.props

    onChangeAddress({
      postalCode: {
        ...address.postalCode,
        visited: true,
      },
    })
  };

  render() {
    const { address, rules } = this.props
    const field = getField('postalCode', rules)

    const loading = !!address.postalCode.loading

    return (
      <InputLabel field={field}>
        <InputText
          className={loading ? 'loading-postal-code' : null}
          field={field}
          address={address}
          onChange={this.handleChange}
          onBlur={this.handleBlur}
        />
        {loading && <PostalCodeLoader />}
      </InputLabel>
    )
  }
}

PostalCode.propTypes = {
  address: PropTypes.shape(AddressShapeWithValidation).isRequired,
  rules: PropTypes.object.isRequired,
  onChangeAddress: PropTypes.func.isRequired,
}

export default PostalCode
