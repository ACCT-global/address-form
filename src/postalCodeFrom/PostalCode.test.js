import React from 'react'
import renderer from 'react-test-renderer'
import PostalCode from './PostalCode'
import { shallow } from 'enzyme'
import BRA from '../country/BRA'
import newAddress from '../__mocks__/newAddress'

describe('Postal Code', () => {
  it('show postal code', () => {
    const address = {
      ...newAddress,
      country: 'BRA',
    }

    const tree = renderer
      .create(
        <PostalCode address={address} rules={BRA} onChangeAddress={jest.fn()} />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should change the postal code', () => {
    const handleChange = jest.fn()
    const wrapper = shallow(
      <PostalCode
        address={{
          ...newAddress,
          country: 'BRA',
          postalCode: '',
        }}
        rules={BRA}
        onChangeAddress={handleChange}
      />
    )

    const event = { target: { value: '2' } }
    wrapper.find('input').simulate('change', event)

    expect(handleChange).toHaveBeenCalledWith({
      ...newAddress,
      country: 'BRA',
      postalCode: '2',
    })
  })
})
