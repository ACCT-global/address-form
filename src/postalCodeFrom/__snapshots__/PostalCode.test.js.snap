// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Postal Code show postal code 1`] = `
<p
  className="input required text"
>
  <label>
    CEP
  </label>
  <input
    className="input-small"
    name="postalCode"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
    value=""
  />
</p>
`;
